packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1537,6 +1544,24 @@ class RenderCustomPaint extends RenderProxyBox {
         newPainter.shouldRepaint(oldPainter)) {
       markNeedsPaint();
     }
+    if (attached) {
```

Comment:
_checkForRepaint sounds like it would be idempotent. Maybe rename this function now that it's a change handler?